packages/arb-token-bridge-ui/tests/e2e/specs/txHistory.cy.ts

```diff
@@ -0,0 +1,147 @@
+import { startWebApp } from '../../support/common'
+
+const DEPOSIT_ROW_IDENTIFIER = /deposit-row-*/i
+const DEPOSIT_SEARCH_IDENTIFIER = /Search for a full or partial L1 tx ID/i
+const WITHDRAWAL_ROW_IDENTIFIER = /withdrawal-row-*/i
+const WITHDRAWAL_SEARCH_IDENTIFIER = /Search for a full or partial L2 tx ID/i
+
+describe('Transaction History', () => {
+  before(() => {
+    cy.changeMetamaskNetwork('goerli')
+    startWebApp()
+  })
+
+  after(() => {
+    cy.logout()
+  })
+
+  // Open tx history panel
+  it('should open transactions history panel', () => {
+    cy.openTransactionsPanel()
+    cy.findByText('Transaction History').should('be.visible')
+  })
+
+  // Test deposits are loading
+  it('should load deposits', () => {
+    cy.findByRole('tab', { name: 'show deposit transactions' })
+      .should('be.visible')
+      .click()
+      .should('have.class', 'selected')
+
+    cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should('have.length.above', 0)
+  })
+
+  // Test deposits pagination works
+  it('should paginate deposits', () => {
+    // check that initially previous button should be disabled
+    cy.findByRole('button', { name: /load previous deposits/i })
+      .should('be.visible')
+      .should('be.disabled')
+
+    // go to next page
+    cy.findByRole('button', { name: /load next deposits/i })
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click()
+    cy.findByText(/page 2/i).should('be.visible')
+    cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should('have.length.above', 0)
+
+    // go to prev page
+    cy.findByRole('button', { name: /load previous deposits/i })
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click()
+    cy.findByText(/page 1/i).should('be.visible')
+    cy.findAllByTestId(DEPOSIT_ROW_IDENTIFIER).should('have.length.above', 0)
+  })
+
+  // Test deposits search works
+  it('should search deposits', () => {
+    // search for valid address substring
+    cy.findByPlaceholderText(DEPOSIT_SEARCH_IDENTIFIER)
+      .type('0x', { scrollBehavior: false })
+      .then(() => {
+        // wait for loader to appear before results
+        cy.wait(2000)
```

Comment:
Do we need this? Cypress should wait for elements to be visible by default.

Reply:
Yeah so it makes sense to use wait here because we are using debounce on the input, and the loader for search only kicks in 1.5 seconds after the last keystroke. If we remove this, the test will mistake the existing rows for search results and will instantaneously pass the test which will be a false positive for search.

Reply:
@chrstph-dvx good point, we could add `.should` here, although I will be merging this branch with mine and we use recursive typing there, which also checks for the correctness of the input

so IMO no point adding it here.